front-end-tools/src/components/ReadComponent.tsx

```diff
@@ -0,0 +1,525 @@
+/* eslint-disable no-console */
+import React, { useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import Select from 'react-select';
+import { Alert, Button, Form, Row } from 'react-bootstrap';
+
+import { WalletConnection } from '@concordium/react-components';
+import {
+    ModuleReference,
+    displayTypeSchemaTemplate,
+    toBuffer,
+    getUpdateContractParameterSchema,
+    ConcordiumGRPCClient,
+} from '@concordium/web-sdk';
+
+import Box from './Box';
+import { read, getEmbeddedSchema, getContractInfo } from '../reading_from_blockchain';
+import { getObjectExample, getArrayExample } from '../utils';
+import { INPUT_PARAMETER_TYPES_OPTIONS } from '../constants';
+
+interface ConnectionProps {
+    account: string;
+    connection: WalletConnection;
+    client: ConcordiumGRPCClient | undefined;
+}
+
+export default function ReadComponenet(props: ConnectionProps) {
+    const { client } = props;
+
+    type FormType = {
+        smartContractIndex: number;
+        smartContractName: string | undefined;
+        entryPointName: string | undefined;
+        file: FileList | undefined;
+        hasInputParameter: boolean;
+        deriveFromSmartContractIndex: boolean;
+        inputParameterType: string | undefined;
+        inputParameter: string | undefined;
+    };
+
+    const form = useForm<FormType>();
+    const deriveContractInfo = form.watch('deriveFromSmartContractIndex');
+    const hasInputParameter = form.watch('hasInputParameter');
+    const entryPointName = form.watch('entryPointName');
+    const smartContractName = form.watch('smartContractName');
+    const inputParameterType = form.watch('inputParameterType');
```

Comment:
It seems to need a controller as well, so would leave it for future investigation/tasks to investigate if we want to introduce controllers or not.

Reply:
I doesn't depend on using `Controller` components, just the `control` object provided by `useForm`, which is just a way to hook the components and functions provided by the library up to the form.

I don't understand your restraint from using this functionality? It's not something that makes anything behave differently. If you want to leave it for later, fair enough, however I fear this will become something that lingers in the backlog.